readthedocs/core/views/serve.py

```diff
@@ -215,3 +215,38 @@ def _serve_symlink_docs(request, project, privacy_level, filename=''):
 
     raise Http404(
         'File not found. Tried these files: %s' % ','.join(files_tried))
+
+
+@map_project_slug
+def robots_txt(request, project):
+    """
+    Serve custom user's defined ``/robots.txt``.
+
+    If the user added a ``robots.txt`` in the "default version" of the project,
+    we serve it directly.
+    """
+    if project.privacy_level == constants.PRIVATE:
+        # If project is private, there is nothing to communicate to the bots.
+        raise Http404()
```

Comment:
Related, what do we do if the project's default version is private? Seems we'll be exposing a something potentially private wihtout this check?

Reply:
Good point.

I think we need to make a decision here:

1. expose the `robots.txt` (I don't really think that this will "expose anything sensitive") --even if your default version is private you will want to communicate what to do with the other ones.
1. disallow the whole site (doesn't make too much sense to me)
1. other?

I'd go for 1).

Reply:
Exposing the `robots.txt` exposes the fact that the project exists, which is definitely a security issue in some cases.

Reply:
Seems safest, especially since it's a new feature. If we get more requests from users we can add more logic here, but doing the safest thing to start feels right.

Reply:
I added more cases, like version is not active or version is not built. Which I think it also makes sense to return 404.